pkg/types/gcp/machinepools.go

```diff
@@ -76,6 +83,9 @@ func (a *MachinePool) Set(required *MachinePool) {
 		}
 		a.EncryptionKey.Set(required.EncryptionKey)
 	}
+	if required.SecureBoot != "" {
```

Comment:
nit: If it happens that it is changed to default to Disabled, this will break. Should we check that this == "Enabled" rather than nothing?

Reply:
The current default is disabled.
Checking for the empty string should be compatible with changing over time default.
If the user specified secureBoot configuration we add it to the machine pool, otherwise we do nothing and get the default.

Reply:
Updated the commit message to make it clearer